### PR TITLE
Update to v0.7.0dev3 and fix docs and tests with it; add CI

### PR DESCRIPTION
**Pull request by brynpickering**
Updated to the latest v0.7 pre-release which allows us to now use this repository as intended!

Namely, we can now update the Calliope model definition schema with our new parameters, so the default values they include also filter through.

I've added continuous integration to run tests. At the moment there is just one integation test: the national-scale example model builds and succesfully solves using the investment steps as an additional dimension.

### CODE REVIEW

**Comment by brynpickering** [comment]
The windows install instructions are a bit of a guess - I don't know that unzipping to the conda prefix `bin` directory will work as expected :P

**Comment by brynpickering** [comment]
python v12 not working because we didn't set up our conda forge recipe correctly: https://github.com/conda-forge/calliope-feedstock/pull/35

Markdown link checks not working because we don't yet have a full build of the documentation to link to.

File: src/calliope_pathways/config/new_param_schema.yaml
Comment (irm-codebase) [inline src/calliope_pathways/config/new_param_schema.yaml:1]: Was this removal intentional?

Comment (brynpickering) [reply src/calliope_pathways/config/new_param_schema.yaml:1]: I moved it to a new subdirectory so that it isn't mixed up with math. I removed one entry from it which is actually not new to pathways: it's already in the pre-defined parameter list.

**Review by irm-codebase — DISMISSED** [review]
